server/src/main/java/org/elasticsearch/action/admin/indices/analyze/TransportAnalyzeAction.java

```diff
@@ -220,10 +220,8 @@ private static Analyzer buildCustomAnalyzer(AnalyzeAction.Request request, Analy
         List<AnalyzeAction.AnalyzeToken> tokens = new ArrayList<>();
         int lastPosition = -1;
         int lastOffset = 0;
-        // Note that we always pass "" as the field to the various Analyzer methods, because
-        // the analyzers we use here are all field-specific and so ignore this parameter
```

Comment:
The upshot of this change is to make the analyze action work in precisely the way this comment says it doesn't... if the analyzer hasn't been built specifically for this request, then we're examining the analyzer for a specific field, and so we use the general index analyzer and pass the field name so that it delegates to the correct field analyzer.